### PR TITLE
RK-5801 - Added maven-perforce instructions

### PR DESCRIPTION
**Pull request by ElDuderinos**
See Readme and pom.xml
Tested locally.

### CODE REVIEW

File: java-maven/maven-perforce/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+regressionTest {
```
Comment (BenTheKlein) [inline java-maven/maven-perforce/Jenkinsfile:1]: you really think its important - it will consume time and resources.....

Comment (ElDuderinos) [reply java-maven/maven-perforce/Jenkinsfile]: you're totally right

File: java-maven/maven-perforce/README.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+     ```bash
+     export ROOKOUT_TOKEN=<Your-Token>
+     ```
+5. Downloading the Rookout Java Agent from available on [maven central]::
+    ```bash
+    $ wget "http://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.rookout&a=rook&v=LATEST"  -O rook.jar
+    ```  
+6. Specifying the Rookout Java Agent when running:
+    ```bash
+    $ java  -javaagent:rook.jar -jar target/rookoutDemo-1.0.0-jar-with-dependencies.jar
```
Comment (LiranLast) [inline java-maven/maven-perforce/README.md:51]: Use `JAVA_TOOL_OPTIONS` :)

File: java-maven/maven-perforce/README.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        <executions>
+            <execution>
+                <phase>validate</phase>
+                <goals>
+                    <goal>create</goal>
+                </goals>
+            </execution>
+        </executions>
+        <configuration>
+            <doCheck>false</doCheck>
```
Comment (LiranLast) [inline java-maven/maven-perforce/README.md:117]: Why false? :)

File: java-maven/maven-perforce/pom.xml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+        <resources>
+            <resource>
+                <directory>${basedir}/src/main/java</directory>
```
Comment (LiranLast) [inline java-maven/maven-perforce/pom.xml:143]: It this source packaging? Why is this here? :)

File: java-maven/maven-perforce/README.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+7. Make sure everything worked: [http://localhost:7000/](http://localhost:7000/hello)
+
+8. Go to [http://app.rookout.com](http://app.rookout.com) and start debugging! 
+
+## Rookout Integration explained
+
+This example is based of the Java javalin "Hello-World" example available [here].
+
+1. We added the source to the project jar(`pom.xml`) to enable rookout:
```
Comment (LiranLast) [inline java-maven/maven-perforce/README.md:62]: What's this?

File: java-maven/maven-perforce/README.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+                <resource>
+                    <directory>${basedir}/src/main/java</directory>
+                </resource>
+            </resources>
+ 
+   ```
+2. In order for Rookout's agent to sync with the right source code version, in this project we udpate the JAR's MANIFEST with the depot path and changelist:
+    ```xml
+    <Rookout-Revision>${buildNumber}</Rookout-Revision>
+    <Rookout-Repository>Perforce:${depotPath}</Rookout-Repository>
```
Comment (LiranLast) [inline java-maven/maven-perforce/README.md:74]: Did you test this with @dan505512? :)

File: java-maven/maven-perforce/README.md (hunk 3 above)
Comment (ElDuderinos) [reply java-maven/maven-perforce/README.md:117]: Do you want us to fail the customer's build process if he has modified files?
Feels intrusive imo, they are grownups, they can fail themselves...

File: java-maven/maven-perforce/pom.xml (hunk 4 above)
Comment (ElDuderinos) [reply java-maven/maven-perforce/pom.xml:143]: Yes, didn't you read the Readme? :)

File: java-maven/maven-perforce/README.md (hunk 5 above)
Comment (ElDuderinos) [reply java-maven/maven-perforce/README.md:62]: source packaging

File: java-maven/maven-perforce/README.md (hunk 6 above)
Comment (ElDuderinos) [reply java-maven/maven-perforce/README.md:74]: not yet

File: java-maven/maven-perforce/README.md (hunk 3 above)
Comment (LiranLast) [reply java-maven/maven-perforce/README.md:117]: Can you add a comment explaining that it can be turned on?

File: java-maven/maven-perforce/pom.xml (hunk 4 above)
Comment (LiranLast) [reply java-maven/maven-perforce/pom.xml:143]: Maybe add a comment here as well marking it is optional if source tagging is used?

File: java-maven/maven-perforce/README.md (hunk 2 above)
Comment (ElDuderinos) [reply java-maven/maven-perforce/README.md]: Done

File: java-maven/maven-perforce/pom.xml (hunk 4 above)
Comment (ElDuderinos) [reply java-maven/maven-perforce/pom.xml:144]: Done
